tools/integration_tests/tokenserver/test_authorization.py

```diff
@@ -517,3 +517,21 @@ def test_x_client_state_must_have_same_client_state_as_key_id(self):
         self.assertEqual(res.json, expected_error_response)
         headers['X-Client-State'] = '616161'
         res = self.app.get('/1.0/sync/1.5', headers=headers)
+
+    def test_x_key_id_header_required(self):
```

Comment:
The new Tokenserver will not accept any requests that do not have an `X-KeyID` header, since it only supports OAuth